### ISSUE TITLE
Unable to install openboard in ubuntu 23.10

### ISSUE THREAD

**Issue by aanandharaja** [OpenBoard-org/OpenBoard#806]
Recently installed ubuntu 23.10, after I tried to install open board 1.7 it shows unmet dependencies, manually installed all files but unable to install these two files libavformat59, libswscale6. can you please rebuild the package for 23.10
Thankyou

**Comment by letsfindaway** [comment]
Which package did you try to install? There is no officially built package for 1.7.0 right now, as this version s still not released.

If you tried it with my experimental packages (https://github.com/letsfindaway/OpenBoard/issues/89): I just enabled builds for 23.10. They need some more minutes to show up in the repository.

**Comment by aanandharaja** [comment]
i tried 1.6.4 and 1.7 from this link https://download.opensuse.org/repositories/home:/letsfindaway:/experimental/xUbuntu_22.10/amd64/, both not working.  latest build 23.10 working fine. Thankyou so much

**Comment by DinoIsanovic** [comment]
![image](https://github.com/OpenBoard-org/OpenBoard/assets/68426922/e9d1e17d-dad4-4098-86c2-d8880f2d76fb)

**Comment by letsfindaway** [comment]
> ![image](https://user-images.githubusercontent.com/68426922/276185811-e9d1e17d-dad4-4098-86c2-d8880f2d76fb.png)

You're are trying to install `openboard_ubuntu_22.04_1.6.4_amd64.deb` on what system? 

Please be aware that you always have to get the package **exactly** matching your distribution's version.

**Comment by DinoIsanovic** [comment]
> You're are trying to install `openboard_ubuntu_22.04_1.6.4_amd64.deb` on what system?
> 
> Please be aware that you always have to get the package **exactly** matching your distribution's version.

on 23.10

**Comment by kaamui** [comment]
the expected dependencies were renamed in 23.10. You can still try to install this package by modifying it with the correct information : 
```
mkdir my_new_folder
mv openboard_ubuntu_..._amd64.deb  my_new_folder/
cd my_new_folder
ar x openboard_ubuntu_...._amd64.deb
tar xf control.tar.xz
*** open the "control" file in a text editor and change the dependencies, then save and exit the text editor ***
tar c postinst prerm md5sums control | xz > control.tar.xz
rm openboard_ubuntu_...._amd64.deb
ar rcs openboard_ubuntu_..._amd64.deb debian-binary control.tar.xz data.tar.xz
```

For example, `libavcodec58` seems to be `libavcodec60` now : https://packages.ubuntu.com/search?keywords=libavcodec

**Comment by letsfindaway** [comment]
Instead of building OpenBoard by yourself, you might also either use the Flatpak or try my [experimental builds](https://github.com/letsfindaway/OpenBoard/issues/89). For Ubuntu 23.10 you can find the packages at https://download.opensuse.org/repositories/home:/letsfindaway:/experimental/xUbuntu_23.10/amd64/

As @kaamui stated above, Ubuntu 23.10 updated `ffmpeg` to a new version and so all the package names related to `ffmpeg` have changed.

**Comment by kaamui** [comment]
> Edit: changing the library name in the package as kaamui explained assumes binary compatibility of the API of both `ffmpeg` versions. I doubt that this can be guaranteed. The package names typically change when the Appilication Binary Interface changes...

Yes, It's why I precised that he could "still try" to do it ^^. It is of course not guaranteed to work, for the reasons you explicited.

**Comment by DinoIsanovic** [comment]
> Instead of building OpenBoard by yourself, you might also either use the Flatpak or try my [experimental builds](https://github.com/letsfindaway/OpenBoard/issues/89). For Ubuntu 23.10 you can find the packages at https://download.opensuse.org/repositories/home:/letsfindaway:/experimental/xUbuntu_23.10/amd64/
> 
> As @kaamui stated above, Ubuntu 23.10 updated `ffmpeg` to a new version and so all the package names related to `ffmpeg` have changed.
> 
> Edit: changing the library name in the package as kaamui explained assumes binary compatibility of the API of both `ffmpeg` versions. I doubt that this can be guaranteed. The package names typically change when the Appilication Binary Interface changes...

THX ... your experimental build is OK except welcome screen :) open board again in use did not tested all functions. I am happy again :)

**Comment by letsfindaway** [comment]
The limited font set is intended: To make documents exchangeable between different computers, OpenBoard restricts the font list to some fonts which are installed together with OpenBoard. So every user of OpenBoard sees the same font list, regardless of platform and system fonts. This avoids many problems when exchanging documents. Did you ever use a special font in a Powerpoint (or LibreOffice Impress) presentation, moved the file to another computer and then the slides looked completely different? OpenBoard wants to avoid this experience.

Currently there is no clean way to extend this list. A dirty workaround is to copy additional fonts to the resource directory of OpenBoard. In my builds, this is `/usr/share/openboard/customizations/fonts`. But note: this folder is owned by the installed package. Copying files to this folder requires root permission and is something you normally should not do, as the package can no longer be cleanly uninstalled without deleting these files manually again. So be warned.

What about the welcome screen? I'm always looking for feedback on my builds. Please describe this problem in detail in a [new issue on my fork](https://github.com/letsfindaway/OpenBoard/issues/new/choose).
